feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/novacard/overview/webViewController/interceptors/NovaCardInterceptor.kt

```diff
@@ -0,0 +1,64 @@
+package io.novafoundation.nova.feature_assets.presentation.novacard.overview.webViewController.interceptors
+
+import android.webkit.CookieManager
+import android.webkit.WebResourceRequest
+import android.webkit.WebResourceResponse
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+
+interface NovaCardInterceptor {
+
+    /**
+     * @return Make a request and return WebResourceResponse if the request was intercepted otherwise null
+     */
+    fun intercept(request: WebResourceRequest): Boolean
+}
+
+fun makeRequest(okHttpClient: OkHttpClient, requestBuilder: Request.Builder): Response {
+    val okHttpResponse = okHttpClient.newCall(requestBuilder.build()).execute()
+
+    if (okHttpResponse.isSuccessful) {
+        return okHttpResponse
+    }
+
+    throw RuntimeException("Request failed with ${okHttpResponse.code} code: ${okHttpResponse.networkResponse?.body}")
+}
```

Comment:
I think it should be an extension for `OkHttpClient`: `OkHttpClient.makeRequestBlocking`